cmd/thanos/sidecar.go

```diff
@@ -321,20 +305,27 @@ func runSidecar(
 }
 
 func validatePrometheus(ctx context.Context, logger log.Logger, m *promMetadata) error {
-	if m.version == nil {
-		level.Warn(logger).Log("msg", "fetched version is nil or invalid. Unable to know whether Prometheus supports /version endpoint, skip validation")
-		return nil
+	flags := promclient.Flags{
```

Comment:
This works, but it might be bit confusing - it's not really following the logical path, which can surprise reader. If there is no flags the whole function should just return. We workaround simple `noFlagEndpoint bool` with assuming some values and unneccesary validating them below - prone to bug in future. But not a blocker, we can merge this and fix it later.